tests/telemetry/test_telemetry.py

```diff
@@ -0,0 +1,56 @@
+# Helper functions
+def get_dict_stdout(gnmi_out, certs_out):
+    """ Extracts dictionary from redis output.
+    """
+    gnmi_list = []
+    gnmi_list = get_list_stdout(gnmi_out) + get_list_stdout(certs_out)
+    # Elements in list alternate between key and value. Separate them and combine into a dict.
+    key_list = gnmi_list[0::2]
+    value_list = gnmi_list[1::2]
+    params_dict = dict(zip(key_list, value_list))
+    return params_dict
+
+def get_list_stdout(cmd_out):
+    out_list = []
+    for x in cmd_out:
+        result = x.encode('UTF-8')
+        out_list.append(result)
+    return out_list
+
+# Test functions
+def test_config_db_parameters(duthost):
+    """Verifies required telemetry parameters from config_db.
+    """
+    gnmi = duthost.shell('/usr/bin/redis-cli -n 4 hgetall "TELEMETRY|gnmi"', module_ignore_errors=False)['stdout_lines']
+    certs = duthost.shell('/usr/bin/redis-cli -n 4 hgetall "TELEMETRY|certs"', module_ignore_errors=False)['stdout_lines']
+    if gnmi is None or certs is None:
+        assert False, "Either TELEMETRY|gnmi or TELEMETRY|certs does not exist in config_db"
+    else:
+        d = get_dict_stdout(gnmi, certs)
+        for key, value in d.items():
+            if str(key) == "client_auth":
```

Comment:
For the existence test, you should check if keys of "client_auth", "ca_crt" exists in the dictionary d. If anyone is missing, the assert should fails.

Reply:
I cant think of scenario of "Telemetry|gnmi" is present but keys under it are not. According to me, all will exist or none. That's why added check at parent level key. This is how we set keys: https://github.com/Azure/sonic-buildimage/pull/3759/files#diff-18b32e4dc045fa3d1c5941a732124c89   Thoughts?

Reply:
discussed offline. agreed on check for each key is not needed because of parent level key check